diff --git a/tea.go b/tea.go
index 0bc915e..6216b6f 100644
--- a/tea.go
+++ b/tea.go
@@ -554,6 +554,13 @@ func (p *Program) Run() (Model, error) {
 		if err != nil {
 			return p.initialModel, err
 		}
+		if runtime.GOOS == "zos" {
+			// Set auto convert on /dev/tty fd for z/OS.
+			// /dev/tty is EBCDIC by default and will send EBCDIC
+			// key strokes to the program if not converted.
+			fd := int(f.Fd())
+			runtime.SetZosAutoConvOnFd(fd, 1047)
+		}
 		defer f.Close() //nolint:errcheck
 		p.input = f
 
